### ISSUE TITLE
[CP] Fix WASM error caused by incorrect return type

### ISSUE THREAD

**Issue by alexmarkov** [dart-lang/sdk#55837]
### Commit(s) to merge

345b4371798e3d035737355421c4d69273b07e52

### Target

beta

### Prepared changelist for beta/stable

https://dart-review.googlesource.com/c/sdk/+/368180

### Issue Description

WebAssembly error
```
Uncaught (in promise) CompileError: WebAssembly.compileStreaming(): Compiling function #1263:"identityHashCode" failed: type error in return[0] (expected i64, got (ref null 103)) @+1479968
```
was caused by the incorrect return type generated by dart2wasm.

### What is the fix

TFA was fixed to avoid inferring nullable return types for methods with non-nullable static return types.

### Why cherry-pick

This bug prevents running an app on WASM.

### Risk

Low

### Issue link(s)

https://github.com/flutter/flutter/issues/148617

**Comment by mkustermann** [comment]
LGTM

**Comment by mkustermann** [comment]
> @athomas @alexmarkov Can we rather cherry-pick this to stable?

Never mind. The `stable` branch is from march (we only started using this TFA information recently).

**Comment by mkustermann** [comment]
As alex is OOO, I've +2'ed the change. This CP has landed on `beta`.

**Comment by athomas** [comment]
Released in 3.5.0-180.2.beta.
